packages/callbacks/example/preview-resolutions/lib/main.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flutter/material.dart';
+import 'dart:ffi';
+import 'dart:isolate';
+
+import 'package:ffi/ffi.dart';
+import 'package:tizen_interop/6.5/tizen.dart';
+import 'package:tizen_interop_callbacks/tizen_interop_callbacks.dart';
+import 'package:tizen_log/tizen_log.dart';
+
+MyAppState? myAppState;
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  const MyApp({super.key});
+
+  @override
+  State<MyApp> createState() => MyAppState();
+}
+
+// A structure which holds data necessary to call the foreach function
+// in another isolate. This includes a SendPort and arguments to foreach call.
+// Same as ForeachCallArgs below, except pointers are kept as integers.
+// We need this to be able to send the pointers to another isolate.
+class ForeachCallArgsAsInts {
+  int cameraPtrInt = 0;
+  int regCallbackInt = 0;
+  int userDataInt = 0;
+  SendPort? sendPort;
+}
+
+// A class which holds foreach call arguments and the SendPort.
+// toInts() converts the data to ForeachCallArgsAsInts().
+class ForeachCallArgs {
+  Pointer<Pointer<camera_cli_s>> cameraPtr = nullptr;
+  Pointer<NativeFunction<Bool Function(Int, Int, Pointer<Void>)>> regCallback =
+      nullptr;
+  Pointer<Void> userData = nullptr;
+  SendPort? sendPort;
+
+  ForeachCallArgs(
+      this.cameraPtr, this.regCallback, this.userData, this.sendPort);
+
+  ForeachCallArgsAsInts toInts() {
+    ForeachCallArgsAsInts res = ForeachCallArgsAsInts();
+
+    res.cameraPtrInt = cameraPtr.address;
+    res.regCallbackInt = regCallback.address;
+    res.userDataInt = userData.address;
+    res.sendPort = sendPort;
+
+    return res;
+  }
+}
+
+class MyAppState extends State<MyApp> {
+  List<String> supportedRes = [];
+  final callbacks = TizenInteropCallbacks();
+
+  @override
+  void initState() {
+    super.initState();
+    myAppState = this;
+    initPlatformState();
+  }
+
+  void initPlatformState() async {
+    const bool exceptionalRetVal = false;
+    int ret;
+
+    Pointer<Pointer<camera_cli_s>> cameraPtr = calloc();
+
+    Log.debug('ConsoleMessage', 'calling tizen.camera_create()');
+    ret = tizen.camera_create(camera_device_e.CAMERA_DEVICE_CAMERA0, cameraPtr);
+    if (ret != 0) {
+      Log.debug('ConsoleMessage', 'tizen.camera_create() failed, ret: $ret');
+      throw Exception('Failed to create camera');
+    }
+
+    Log.debug('ConsoleMessage', 'calling register()');
+    final rcbPreview =
+        callbacks.register<Bool Function(Int, Int, Pointer<Void>)>(
+            'camera_supported_preview_resolution_cb',
+            Pointer.fromFunction(dartPreviewResCb, exceptionalRetVal),
+            nullptr,
+            true);
+
+    final recPortPreview = ReceivePort();
+
+    // Prepare foreach call's arguments that will be sent to the spawned isolate.
+    ForeachCallArgs args = ForeachCallArgs(cameraPtr, rcbPreview.regCallbackPtr,
+        rcbPreview.regUserData, recPortPreview.sendPort);
```

Comment:
What do `rcb` and `rec` stand for? You may avoid abbreviations as much as possible. Please consider using better naming that can be easily understood by readers who read the code for the first time throughout this file.